File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program11.java
import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public void Display()
	{
		if(iValue<0)
                {
                	iValue = -iValue;
                }
                        
		while(iValue!=0)
                {
			int iDigit = iValue%10;  
			System.out.println(iDigit);
        	  	iValue = iValue/10;
          
  		}
	}
}

class Program1
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		obj.Display();
	}
}

/*
OUTPUT:

Enter number: 
2395
5
9
3
2

*/

File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program11.cpp
#include<iostream>
using namespace std;

class Display
{
	private:
		int iValue;

	public:

		Display(int iValue)
		{
			this->iValue = iValue;
		}	

		void DisplayDigit()
		{
			if(iValue<0)
			{
				iValue = -iValue;
			}
			while(iValue!=0)
			{
				int iDigit = iValue%10;
				cout<<iDigit<<"\n";
				iValue = iValue/10;
			}
		}
};

int main()
{
	int iValue=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	dObj->DisplayDigit();

	return 0;
}

/*
OUTPUT:

Enter Number: 2395
5
9
3
2


Enter Number: -1018
8
1
0
1

*/
